merlin/schema/io/tensorflow_metadata.py

```diff
@@ -35,69 +35,138 @@
 
     def to_proto_text(self) -> str:
+        """Convert this TensorflowMetadata schema object to a Proto text string
+
+        Returns:
+            str: Generated Proto text string
+        """
         from tensorflow_metadata.proto.v0 import schema_pb2
 
         return proto_utils.better_proto_to_proto_text(self.proto_schema, schema_pb2.Schema())
 
-    def to_proto_text_file(self, path: str, file_name="schema.pbtxt") -> "TensorflowMetadata":
-        return _write_file(self.to_proto_text(), path, file_name)
+    def to_proto_text_file(self, path: str, file_name="schema.pbtxt"):
+        """Write this TensorflowMetadata schema object to a file as a Proto text string
+
+        Args:
+            path (str): Path to the directory containing the Protobuf text file
+            file_name (str, optional): Name of the output file. Defaults to "schema.pbtxt".
+        """
+        _write_file(self.to_proto_text(), path, file_name)
 
     def copy(self, **kwargs) -> "TensorflowMetadata":
+        """Create a copy of this TensorflowMetadata schema object
+
+        Returns:
+            TensorflowMetadata: Copy of this TensorflowMetadata schema object
+        """
         schema_copy = proto_utils.copy_better_proto_message(self.proto_schema, **kwargs)
         return TensorflowMetadata(schema_copy)
 
     @classmethod
-    def from_merlin_schema(cls, schema: MerlinSchema):
+    def from_merlin_schema(cls, schema: MerlinSchema) -> "TensorflowMetadata":
+        """Convert a MerlinSchema object to a TensorflowMetadata schema object
+
+        Args:
+            schema (MerlinSchema): Schema object to convert
+
+        Returns:
+            TensorflowMetadata: Schema converted to a TensorflowMetadata schema object
+        """
         features = []
         for col_name, col_schema in schema.column_schemas.items():
-            features.append(pb_feature(col_schema))
+            features.append(_pb_feature(col_schema))
 
         proto_schema = ProtoSchema(feature=features)
 
         return TensorflowMetadata(proto_schema)
 
-    def to_merlin_schema(self):
+    def to_merlin_schema(self) -> MerlinSchema:
+        """Convert this TensorflowMetadata schema object to a MerlinSchema object
+
+        Returns:
+            MerlinSchema: Schema converted to MerlinSchema object
+        """
         merlin_schema = MerlinSchema()
 
         for feature in self.proto_schema.feature:
-            col_schema = merlin_column(feature)
+            col_schema = _merlin_column(feature)
             merlin_schema.column_schemas[col_schema.name] = col_schema
 
         return merlin_schema
 
-    def to_json(self):
+    def to_json(self) -> str:
+        """Convert this TensorflowMetadata schema object to a JSON string
+
+        Returns:
+            str: Schema converted to a JSON string
+        """
         return self.proto_schema.to_json()
 
 
-def pb_int_domain(column_schema):
+def _pb_int_domain(column_schema):
```

Comment:
Semi-private methods that start with a leading underscore are exempted from the docstring requirement, and these don't appear to be called from anywhere else so I made them semi-private.

(Over the long haul, we may also want to add docstrings to semi-private methods, but trying to keep the scope manageable for 22.04.)

Reply:
I think thats a good callout, and we should get in the habit of separating out public/private API's